util/db/dbutil.go

```diff
@@ -125,24 +127,30 @@ func Retry(fn func() error) (err error) {
 	}
 }
 
-// Atomic executes a piece of code with respect to the database atomically.
-// For transactions where readOnly is false, sync determines whether or not to wait for the result.
-func (db Accessor) Atomic(fn idemFn) (err error) {
-	descr := "w"
-	if db.readOnly {
-		descr = "r"
+// getDecoratedLogger retruns a decorated logger that includes the readonly true/false, caller and extra fields.
+func (db *Accessor) getDecoratedLogger(fn idemFn, extras ...interface{}) logging.Logger {
+	log := logging.Base().With("readonly", db.readOnly)
+	_, file, line, ok := runtime.Caller(2)
+	if ok {
+		log = log.With("caller", fmt.Sprintf("%s:%d", file, line))
+	}
+	log = log.With("callee", runtime.FuncForPC(reflect.ValueOf(fn).Pointer()).Name())
+	if len(extras) > 0 {
+		for i, e := range extras {
+			if e == nil || reflect.ValueOf(e).IsNil() {
+				continue
```

Comment:
Does our logging package panic on a nil `With()` value?

Reply:
no, but it seems to printing out [null] when we're not passing any arguments, so I figured that I'll drop these non-useful fields. ( while retaining the index, so the n-th `nil` could be figured out )

Reply:
Fair enough, it does seem like the log message has enough context to figure out what the caller extras args were, even for nil values.